gh-107017: Change Chapter Strings to Texts in the Introduction chapter.

TommyUnreal opened the pull request:
While the term "string" is well-known in computer science, it might not resonate with beginners or non-technical individuals. Term Texts emphasize practical application in a similar way to Numbers and Lists chapters. This can make the introduction part less intimidating and more beginner-oriented. On the other hand, people coming from other technologies will not be surprised that text is represented with strings.

<!-- gh-issue-number: gh-107017 -->
* Issue: gh-107017
<!-- /gh-issue-number -->


<!-- readthedocs-preview cpython-previews start -->
----
:books: Documentation preview :books:: https://cpython-previews--107104.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

TommyUnreal committed:
Change Strings to Texts in introduction chapter.

While term "string" is well known in computer science, it might not resonate with beginners or non-technical individuals. Term Texts emphasize practical application in similar way to Numbers and Lists chapters. This can make the introduction part less intimidating and more beginner-oriented. On the other hand people comming from other technologies will not be surprised that text is represented with strings.

evildmp commented on Doc/tutorial/introduction.rst:
```diff
@@ -138,13 +138,14 @@ and uses the ``j`` or ``J`` suffix to indicate the imaginary part
 .. _tut-strings:
 
-Strings
+Texts
```
I'd actually keep the heading _Strings_, because that's the correct name.

evildmp commented on Doc/tutorial/introduction.rst:
```diff
@@ -138,13 +138,14 @@ and uses the ``j`` or ``J`` suffix to indicate the imaginary part
-in several ways.  They can be enclosed in single quotes (``'...'``) or
-double quotes (``"..."``) with the same result [#]_.  ``\`` can be used
-to escape quotes::
+Different kinds of text have the type :class:`str`.  This includes
```
I would suggest:

> Python can manipulate text ("strings") as well as numbers. This can include [your list of examples]. 

> Strings are enclosed [...].

evildmp commented on Doc/tutorial/introduction.rst:
```diff
@@ -138,13 +138,14 @@ and uses the ``j`` or ``J`` suffix to indicate the imaginary part
+characters "``!``", words "``rabbit``", names "``Paris``", sentences
+"``Got your back.``", etc. "``Yay! :)``".  They can be enclosed in single
+quotes (``'...'``) or double quotes (``"..."``) with the same result [#]_.
+``\`` can be used to escape quotes::
```
I think the concept of escaping (and the word) is not familiar to many beginners, so I'd suggest something like:

> To quote a quote, we need to "escape" it, by preceding it with ``\``::

evildmp commented on Doc/tutorial/introduction.rst:
```diff
@@ -138,13 +138,14 @@ and uses the ``j`` or ``J`` suffix to indicate the imaginary part
-to escape quotes::
+Different kinds of text have the type :class:`str`.  This includes
+characters "``!``", words "``rabbit``", names "``Paris``", sentences
+"``Got your back.``", etc. "``Yay! :)``".  They can be enclosed in single
```
Perhaps add: 

> Even "123" is a string (and not a number - because it's enclosed in quotes.

TommyUnreal committed:
New structure of first paragraph in Text chapter.

TommyUnreal committed:
Refactor string quotes handling and examples.

Changed the description, removed duplicate explanations and reworked
examples so they are more consice with its paragraph.

evildmp commented on Doc/tutorial/introduction.rst:
```diff
@@ -160,23 +168,15 @@ quotes (``'...'``) or double quotes (``"..."``) with the same result [#]_.
-strings are equivalent.  The string is enclosed in double quotes if
-the string contains a single quote and no double quotes, otherwise it is
-enclosed in single quotes.  The :func:`print` function produces a more
+In the interactive interpreter, the string definition and output string
```
"In the Python shell..."

TommyUnreal committed:
Replaced interactive interpreter with Python shell

hugovk commented on Doc/tutorial/introduction.rst:
```diff
@@ -138,16 +138,25 @@ and uses the ``j`` or ``J`` suffix to indicate the imaginary part
 
-Strings
+Text
 -------
```
Adjust length to match header:
```suggestion
----
```

hugovk commented on Doc/tutorial/introduction.rst:
```diff
@@ -138,16 +138,25 @@ and uses the ``j`` or ``J`` suffix to indicate the imaginary part
-in several ways.  They can be enclosed in single quotes (``'...'``) or
-double quotes (``"..."``) with the same result [#]_.  ``\`` can be used
-to escape quotes::
+Python can manipulate text (represented by type :class:`str`, so called
```
```suggestion
Python can manipulate text (represented by type :class:`str`, so-called
```

hugovk commented on Doc/tutorial/introduction.rst:
```diff
@@ -138,16 +138,25 @@ and uses the ``j`` or ``J`` suffix to indicate the imaginary part
 
    >>> 'spam eggs'  # single quotes
    'spam eggs'
+   >>> "Paris rabbit got your back :)! Yay!" # double quotes
```
Two spaces before the hash to follow PEP 8 and match the others here:

```suggestion
   >>> "Paris rabbit got your back :)! Yay!"  # double quotes
```

hugovk commented on Doc/tutorial/introduction.rst:
```diff
@@ -138,16 +138,25 @@ and uses the ``j`` or ``J`` suffix to indicate the imaginary part
    'spam eggs'
+   >>> "Paris rabbit got your back :)! Yay!" # double quotes
+   'Paris rabbit got your back :)! Yay!'
+   >>> '1975' # digits and numerals enclosed in quotes are also strings
```
```suggestion
   >>> '1975'  # digits and numerals enclosed in quotes are also strings
```

hugovk reviewed (changes requested):
Some minor suggestions, thanks for the PR!

TommyUnreal committed:
Adjust length to match header

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

TommyUnreal committed:
Fix grammar.

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

TommyUnreal committed:
Add a space before the hash to follow PEP 8

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

TommyUnreal committed:
Add a space before the hash to follow PEP 8

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

hugovk reviewed (approved):
Thank you!
